applications/tari_base_node/src/parser.rs

```diff
@@ -175,4 +178,82 @@ impl Parser {
             },
         }
     }
+
+    // Function to process  the get balance command
+    fn process_get_balance(&mut self) {
+        let mut handler = self.base_node_context.wallet_output_service.clone();
+        self.executor.spawn(async move {
+            let result = handler.get_balance().await;
+            if result.is_err() {
+                println!("Something went wrong");
+                warn!(
+                    target: LOG_TARGET,
+                    "Error communicating with wallet: {}",
+                    result.err().unwrap().to_string(),
+                );
+                return;
+            };
+            println!("Current balance is: {:?}", result.unwrap());
+        });
+    }
+
+    // Function to process  the get chain meta data
+    fn process_get_chain_meta(&mut self) {
+        let mut handler = self.base_node_context.node_service.clone();
+        self.executor.spawn(async move {
+            let result = handler.get_metadata().await;
+            if result.is_err() {
+                println!("Something went wrong");
+                warn!(
+                    target: LOG_TARGET,
+                    "Error communicating with base node: {}",
+                    result.err().unwrap().to_string(),
+                );
+                return;
+            };
+            println!("Current meta data is is: {:?}", result.unwrap());
+        });
+    }
+
+    // Function to process  the send transaction function
+    fn process_send_tari(&mut self, command_arg: Vec<&str>) {
+        if command_arg.len() != 3 {
+            println!("Command entered wrong, please enter in the following format: ");
+            println!("send_tari [amount of tari to send] [public key to send to]");
+            return;
+        }
+        let amount = command_arg[1].parse::<u64>();
+        if amount.is_err() {
+            println!("please enter a valid amount of tari");
+            return;
+        }
```

Comment:
The `match` pattern is more concise and more idiomatic.

Reply:
done